plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java

```diff
@@ -120,7 +134,7 @@ public Page getNextPage()
 
             Optional<RowPredicate> deleteFilterPredicate = deletePredicate.get();
             if (deleteFilterPredicate.isPresent()) {
-                dataPage = deleteFilterPredicate.get().filterPage(dataPage);
+                dataPage = deleteFilterPredicate.get().filterPage(dataPage, deleteFilterIndexes);
```

Comment:
I don't think you need to change interface, you can pass in `deleteFilterIndexes.length == 0 ? dataPage : dataPage.getColumns(deleteFilterIndexes)` to existing method

Reply:
The `dataPage` contains non-filter columns. Calling `getColumns` early results in out-of-index failure.

Reply:
I'm not sure I follow, if calling `getColumns` works inside `filterPage` then it should work here too as the input page or deleteFilterIndexes are not changing

Reply:
`getColumns` is used only for `test()` call in `filterPage`. Your suggestion affects all `page` in `filterPage`.